elisa/src/client/RunDetails/RunDetails.tsx

```diff
@@ -0,0 +1,245 @@
+        const controls = getUniqueIdsForPlotSelections(props.data, plotOptions, CONTROL_COL_NAME);
+
+        this.state = {
+            filteredData: filterDataByPlotOptions(props.data, samples, controls, plotOptions),
+            plotOptions,
+            plates,
+            spots,
+            samples,
+            controls,
+            curveFitData: undefined
+        };
+    }
+
+    componentDidMount(): void {
+        this.getCurveFitData();
+    }
+
+    componentDidUpdate(prevProps: Readonly<ImplProps>, prevState: Readonly<ImplState>, snapshot?: any): void
+    {
+        if (shouldReloadCurveFitData(prevState.plotOptions, this.state.plotOptions)) {
+            this.getCurveFitData();
+        }
+    }
+
+    getCurveFitData() {
+        const { protocolId, runId, data } = this.props;
+        const { plotOptions } = this.state;
+        const filteredData = filterDataByPlotOptions(data, [], [], plotOptions, false);
+
+        getCurveFitXYPairs(protocolId, runId, plotOptions.plateName, plotOptions.spot, filteredData)
+            .then(curveFitData => {
+                this.setState(() => ({ curveFitData }));
+            })
+            .catch(reason => {
+                console.error(reason);
+                this.setState(() => ({ curveFitData: {
+                    points: [],
+                    error: reason.exception
+                } as CurveFitData }));
+            });
+    }
+
+    setPlotOption = (key: string, value: any, resetIdSelection: boolean) => {
+        const { data } = this.props;
+        const { plotOptions, samples, controls, curveFitData } = this.state;
+        const updatedPlotOptions = getUpdatedPlotOptions(key, value, resetIdSelection, plotOptions);
+
+        this.setState(() => ({
+            plotOptions: updatedPlotOptions,
+            filteredData: filterDataByPlotOptions(data, samples, controls, updatedPlotOptions),
+            samples: getUniqueIdsForPlotSelections(data, updatedPlotOptions, SAMPLE_COL_NAME),
+            controls: getUniqueIdsForPlotSelections(data, updatedPlotOptions, CONTROL_COL_NAME),
+            curveFitData: shouldReloadCurveFitData(plotOptions, updatedPlotOptions) ? undefined : curveFitData
+        }));
+    };
+
+    render() {
+        const { columnInfo } = this.props;
+        const { filteredData } = this.state;
+        const missingFieldKeys = getMissingFields(columnInfo);
+
+        return (
+            <>
+                {missingFieldKeys?.length > 0 &&
+                    <Alert bsStyle={'warning'}>
+                        Warning: the assay design is missing the following fields which may affect the various plotting
+                        features and display of this page: {missingFieldKeys.join(', ')}.
+                    </Alert>
+                }
+                <div className={'row'}>
+                    <div className={'col col-xs-12 col-md-3 run-details-left'}>
```

Comment:
I might be misunderstanding the .col class but thought that was more for when you don't have a specific col size class (col-xs-12) on a column.

Reply:
You are right, the 'col' isn't needed here. Removed.